CONTRIBUTING.md

```diff
@@ -113,7 +113,161 @@ You can use `git rebase --interactive` to reorganize your commits and  `git
 push --force`  to update your branch. Depending on your IDE, there are various
 features or plugins to streamline this process (e.g.,
 [vim-fugitive](https://github.com/tpope/vim-fugitive),
-[magit](https://magit.vc/)).
+[magit](https://magit.vc/)). Git also provides `git commit --fixup` and
+`git rebase --autosquash` which can streamline this process.
+
+# Build and Release Versions
+
+Release versions of Katana are a semantic version with an release candidate tag
+(`rc#`) appended for release candidates.
+
+Build/development versions of Katana are are in the form:
+`[release].dev+N.0.######`.
+
+* a release version followed
+* `.dev`
+* `+`
+* the number of commits on the nearest major branch (e.g., `master` or `release/v*`)
+* `0`
+* an abbreviated commit id
+
+The `0` is to leave space for the enterprise commit count in enterprise builds
```

Comment:
"to leave space" -> "a placeholder"? Should we be even more explicit and say that for enterprise releases only it is replaced with the enterprise repo commit count?